miio/walkingpad.py

```diff
@@ -0,0 +1,189 @@
+
+    @property
+    def step(self) -> int:
+        """Current steps."""
+        return self.data["step"]
+
+    @property
+    def distance(self) -> int:
+        """Current distance."""
+        return self.data["dist"]
+
+    @property
+    def calories(self) -> int:
+        """Current distance."""
+        return self.data["cal"]
+
+
+class Walkingpad(Device):
+    """Main class representing Xiaomi Walkingpad."""
+
+    # TODO: - Auto On/Off Not Supported
+    #       - Adjust Scenes with Wall Switch Not Supported
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode}\n"
+            "Time: {result.time}\n"
+            "Steps: {result.step}\n"
+            "Speed: {result.speed}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}  ",
+        )
+    )
+    def status(self) -> WalkingpadStatus:
+        """Retrieve properties."""
+
+        properties_received = []
+        values_received = []
+
+        # Walkingpad A1 allows you to retrieve a subset of values with "all"
+        # eg ['mode:1', 'time:1387', 'sp:3.0', 'dist:1150', 'cal:71710', 'step:2117']
+
+        properties = ["all"]
+
+        # Walkingpad A1 only allows 1 property to be read at a time
+
+        values = self.get_properties(properties, max_properties=1)
+
+        # When running the tests, for some reason the list provided is passed within another list, so I take
+        # care of this here.
+
+        if any(isinstance(el, list) for el in values):
+            values = values[0]
+
+        for x in values:
+            prop_split, value = x.split(":")
+            properties_received.append(prop_split)
+            values_received.append(value)
+
+        properties_additional = ["power", "mode"]
+
+        values_additional = self.get_properties(properties_additional, max_properties=1)
+
+        properties_received = properties_received + properties_additional
+        values_received = values_received + values_additional
+        return WalkingpadStatus(
+            defaultdict(lambda: None, zip(properties_received, values_received))
+        )
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_power", ["on"])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    @command(default_output=format_output("Starting the treadmill"))
+    def start(self):
+        """Starting Up."""
+        return self.send("set_state", ["run"])
+
+    @command(default_output=format_output("Stopping the treadmill"))
+    def stop(self):
+        """Starting Up."""
+        return self.send("set_state", ["stop"])
+
+    @command(
+        click.argument("mode", type=int),
+        default_output=format_output("Setting mode to {mode}"),
+    )
+    def set_mode(self, mode: int):
```

Comment:
Please use an enum for the mode setting.

Reply:
Done